master/internal/resourcemanagers/fair_share.go

```diff
@@ -50,11 +50,41 @@ func (g groupState) String() string {
 }
 
 func (f *fairShare) Schedule(rp *ResourcePool) ([]*sproto.AllocateRequest, []*actor.Ref) {
+	defer f.updateJobs(rp) // CHECK does this work how I think it does?
 	return fairshareSchedule(rp.taskList, rp.groups, rp.agents, rp.fittingMethod)
 }
 
+func (f *fairShare) createJobQInfo(
+	rp *ResourcePool,
+) (job.AQueue, map[model.JobID]*actor.Ref) {
+	for _, resAllocated := range rp.taskList.allocations {
+		req := rp.taskList.taskByID[resAllocated.ID]
+		if req.JobID == nil {
+			continue
+		}
+		req.State = job.SchedulingStateScheduled
+	}
+	reqs := make(AllocReqs, 0)
+	for _, req := range rp.taskList.taskByID {
+		reqs = append(reqs, req)
+	}
+	jobQ, jobActors := mergeToJobQInfo(reqs)
+	for _, j := range jobQ {
+		j.JobsAhead = -1 // we don't support job order for fairshare scheduler.
```

Comment:
would that accurately represent the order in which the jobs get scheduled? if so we should do that otherwise I'd stick with just going off of the job id to favor correctness and clarity 🤔

Reply:
> Or is this a functionality we can provide in the webui side?

due to pagination, we'd probably need to do it server-side but I think we can do it easily in a separate pr. I'd say we move ahead as is to get a baseline job representation for fairshare first.